packages/plugin/vite/src/config/vite.base.config.ts

```diff
@@ -79,14 +82,15 @@ export function pluginHotRestart(command: 'reload' | 'restart'): Plugin {
     name: '@electron-forge/plugin-vite:hot-restart',
     closeBundle() {
       if (command === 'reload') {
-        for (const server of Object.values(process.viteDevServers)) {
+        for (const server of Object.values(viteDevServers)) {
           // Preload scripts hot reload.
           server.ws.send({ type: 'full-reload' });
         }
-      } else {
+      } else if (command === 'restart') {
         // Main process hot restart.
         // https://github.com/electron/forge/blob/v7.2.0/packages/api/core/src/api/start.ts#L216-L223
-        process.stdin.emit('data', 'rs');
+        // TODO: blocked in #3380
```

Comment:
I observe that what you replied in this PR has already been implemented in another PR. So, should we address this TODO?

Reply:
Yes, I will go ahead and implement this feature.